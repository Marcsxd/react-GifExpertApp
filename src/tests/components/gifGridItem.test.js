import React from 'react'
import { shallow } from 'enzyme'

import GifGridItem from '../../components/GifGridItem'


describe('Pruebas en el componente gifGridItem', () => {

    const title = 'Titulo random';
    const url = 'https://NoTeCreoHijo.localhost.jpg'
    const wrapper = shallow( <GifGridItem title = { title } url = { url } /> )


    test('debe mostrar correctamente el componente', () => {

        expect( wrapper ).toMatchSnapshot()

    })

    test('debe tener un pÃ¡rrafo con el title', () => {

        const p = wrapper.find('p')

        expect( p.text().trim() ).toBe( title )

    })

    test('la img debe tener los mismos atributos que los props', () => {

        const img = wrapper.find('img');

        // console.log(img.props().src);   console.log(img.prop('src'));

        expect( img.prop('src')).toBe( url );
        expect( img.prop('alt')).toBe( title );
        
    })

    test('el div debe tener la clase animate__fadeIn', () => {

        const div = wrapper.find( 'div' );
        const classNames = div.prop('className');

        expect( classNames.includes('animate__fadeIn') ).toBe(true)

    })

})

