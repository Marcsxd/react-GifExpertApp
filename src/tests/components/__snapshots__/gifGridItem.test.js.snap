// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente gifGridItem debe mostrar correctamente el componente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Titulo random"
    src="https://NoTeCreoHijo.localhost.jpg"
  />
  <p>
     
    Titulo random
     
  </p>
</div>
`;
